From dc54b0eb70110e9ada41243705d0a55d55f0a3d4 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Thu, 2 Nov 2023 23:48:53 +0100
Subject: [PATCH 25/26] arm64: dts: imx8mp: Describe CSI2 GPIO expander on
 i.MX8MP DHCOM PDK3 board

The production PDK3 carrier board rev.200 contains additional GPIO
expander to control power and reset signals for each CSI2 plug
separately. Describe this expander in the carrier board DT. The
label is used by sensor DTOs to reference the expander and its
signals.

Upstream-Status: Backport [a196684376d51bf6b1d2ad6542f5ea17f243fe82]
Signed-off-by: Marek Vasut <marex@denx.de>
Signed-off-by: Shawn Guo <shawnguo@kernel.org>
---
 arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts
index ead474108cfb1..407caac29ac17 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts
@@ -179,6 +179,16 @@ sgtl5000: codec@a {
 				VDDIO-supply = <&reg_vdd_3p3v_awo>;
 			};
 
+			csi2exp: gpio@24 {
+				compatible = "nxp,pca9570";
+				reg = <0x24>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				gpio-line-names =
+					"CSI2_#RESET", "CSI2_#PWDN",
+					"CSI_#PWDN", "CSI_#RESET";
+			};
+
 			typec@3d {
 				compatible = "nxp,ptn5150";
 				reg = <0x3d>;
-- 
2.43.0

